ma_gym/envs/traffic_junction/traffic_junction.py

```diff
@@ -44,20 +47,47 @@ class TrafficJunction(gym.Env):
     """
     metadata = {'render.modes': ['human', 'rgb_array']}
 
-    def __init__(self, grid_shape=(14, 14), n_agents=10, n_max=10, arrive_prob=0.5, full_observable=False):
+    def __init__(self, grid_shape=(14, 14), step_cost=0, n_max=4, rcoll=-10, arrive_prob=0.5, full_observable=False):
         self._grid_shape = grid_shape
-        self.n_agents = n_agents
+        self.n_agents = n_max
         self._max_steps = 100
-        self._step_count = None
-        self._collision_reward = -10
+        self._step_count = 0
+        self._collision_reward = rcoll
         self._total_episode_reward = None
         self._arrive_prob = arrive_prob
         self._n_max = n_max
+        self._step_cost = step_cost
+        self.curr_cars_count = 0
 
-        self._entry_gates = [(7, 0), (13, 7), (0, 6), (6, 13)]
+        self._agent_view_mask = (3, 3)
+        mask_size = np.prod(self._agent_view_mask)
+
+        # entry gates where the cars spawn
+        self._entry_gates = [(self._grid_shape[0] // 2, 0), (self._grid_shape[0] - 1, self._grid_shape[1] // 2),\
```

Comment:
The order of cars should be either clockwise or anti-clockwise.

The correct order would be : [(7, 0), (13, 7), (6, 13), (0, 6)]

Reply:
Also, I guess, changing the 'self._entry_gates' would require changes in `self._destination`